compiler/damlc/tests/BUILD.bazel

```diff
@@ -227,7 +227,7 @@ damlc_compile_test(
     srcs = [":bond-trading"],
     heap_limit = "200M" if is_windows else "100M",
     main = "bond-trading/Test.daml",
-    stack_limit = "35K",
+    stack_limit = "300K" if is_windows else "35K",
```

Comment:
Please don’t simply bump these limits. That defeats the whole purpose of these tests. If we need to bump the stack limit (even if just on Windows), there is a reason for it and at the very least we need to figure out what it is before we blindly bump the limit and accept regressions.

Reply:
I suppose the reason why we need more stack space now is that generate more code during the template desugaring. The bump to `300K` seems excessive though. @rohanjr can you please shrink it to something more reasonable, like `40K` or `50K`.

Reply:
I could see higher heap usage originating from generating more code but higher stack usage does at least not necessarily seem related and is often a sign of spaceleaks.